client/src/components/CreatePoll/index.js

```diff
@@ -0,0 +1,287 @@
+import Fade from '@material-ui/core/Fade';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import Typography from '@material-ui/core/Typography';
+
+import UserContext from '../../context/UserContext';
+
+export default function CreatePoll ({ openCreatePoll, setOpenCreatePoll }) {
+    const { userData } = useContext(UserContext);
+    
+    const initialPoll = {
+        question: '', 
+        friendlist: 'List 1',
+        error: null,
+        userId: userData.user.id,
+    };
+
+    const friendLists = [
+        {
+          value: 'List 1',
+        },
+        {
+          value: 'List 2',
+        },
+        {
+          value: 'List 3',
+        },
+      ];
+
+    const [newPoll, setNewPoll] = useState(initialPoll);
+    const [friendList, setFriendList] = useState('List 1');
+    const [fileInputStateOne, setFileInputStateOne] = useState('');
+    const [fileInputStateTwo, setFileInputStateTwo] = useState('');
+    const [previewSourceOne, setPreviewSourceOne] = useState();
+    const [previewSourceTwo, setPreviewSourceTwo] = useState();
+
+    const handleClose = () => {
+        setOpenCreatePoll(false);
+    };
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setNewPoll(prevPoll => {
+            return { 
+                ...prevPoll, 
+                [name]: value, 
+            }
+        });
+    };
+
+    const handleSelect = (e) => {
+        const { name, value } = e.target;
+        setNewPoll(prevPoll => {
+            return { 
+                ...prevPoll, 
+                [name]: value, 
+            }
+        });
+
+        setFriendList(e.target.value);
+    };
+
+    const handleFileInputChangeOne = (e) => {
+        const file = e.target.files[0];
+        previewFileOne(file);
+    };
+
+    const previewFileOne = (file) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onloadend = () => {
+            setPreviewSourceOne(reader.result);
+        };
+    };
+
+    const handleFileInputChangeTwo = (e) => {
+        const file = e.target.files[0];
+        previewFileTwo(file);
+    };
+
+    const previewFileTwo = (file) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onloadend = () => {
+            setPreviewSourceTwo(reader.result);
+        };
+    };
+    
+    const handleSubmitPoll = (e) => {
+        e.preventDefault();
+        if(!newPoll) return;
+        uploadImage(previewSourceOne, previewSourceTwo, newPoll);
+        setNewPoll(initialPoll);
+        setPreviewSourceOne('');
+        setPreviewSourceTwo('');
+        setOpenCreatePoll(false);
+    };
+
+    const uploadImage = async (previewSourceOne, previewSourceTwo, newPoll) => {
+        try {
+            await axios.post('http://localhost:5000/polls/upload', {
```

Comment:
eventually we'll have to put the server domain `http://localhost:5000/` into a config variable